### ISSUE TITLE
missing hostname line in config generated on screen by "rclone authorize " 

### ISSUE THREAD

**Issue by sylvainLS** [rclone/rclone#4493]
#### What is the problem you are having with rclone?
i have used "rclone authorize " to generate the config file on my windows PC to use it after on a light unix server .
I get only 2 generated lines on the screen
1/ the remote label
2/ the token

it misses the hostname.

when I have used  "rclone config " on my windows PC , all is OK.
the config file contains 3 lines
1/ the remote label
2/ the hostname
3/ the token

I copy the config file on my unix server and all works perfectly.

It is a small bug, but , as a newbie , I spend one hour to understand the issue.
there is no example of the config in the doc I red to check the completeness.

#### What is your rclone version (output from `rclone version`)
rclone-v1.52.3-windows-amd64

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
windows 10  , 64 bits

####  Which cloud storage system are you using? (eg Google Drive)
pcloud

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
"rclone authorize "

**Comment by animosity22** [comment]
If you use rclone authorize, you don't need a rclone.conf on the machine running it. You just need to authorize the cloud provider and paste the code into the headless machine.

So:

```
rclone authorize pcloud
```

and that's it.

**Comment by animosity22** [comment]
It also says this in the rclone config process to show you:

```
y/n> N
For this to work, you will need rclone available on a machine that has
a web browser available.

For more help and alternate methods see: https://rclone.org/remote_setup/

Execute the following on the machine with the web browser (same rclone
version recommended):

	rclone authorize "pcloud"

Then paste the result below:
```

**Comment by animosity22** [comment]
This was answered so I'm closing it out.

If you have a follow up question, please use the forums https://forum.rclone.org

**Comment by sylvainLS** [comment]
hi,
I think you didn't understand well my first post.
sorry if I am not clear and thank you for your work on rclone.

when I use  rclone authorize "pcloud", the result printed on screen is incomplete !!!!
it misses the " the hostname" on the sceen ( a little oversight in the code I think ).
please try to reproduce it and you will understand.
If I pasted the config lines without "hostname" on the headless machine   ( what I have done first as told in the doc : Then paste the result below: )  , it obviously fails.

I find the solution by myself. my post is only for future users

**Comment by sylvainLS** [comment]
on my desktop PC windows 10
![rclone](https://user-images.githubusercontent.com/69429566/90803566-3882b880-e319-11ea-98c9-a28f7b922d69.png)

on remote headless machine (NAS)
rclone v1.52.3
- os/arch: linux/amd64
- go version: go1.14.7
<img width="498" alt="rclone2" src="https://user-images.githubusercontent.com/69429566/90803680-68ca5700-e319-11ea-8773-71daef29032d.png">
=> the generated config file is  not OK : it misses the hostname

so it seems the issue is not on the desktop windows PC but on the auto config onthe headless machine.

it shoud be 
<img width="193" alt="rclone3" src="https://user-images.githubusercontent.com/69429566/90804006-e55d3580-e319-11ea-90be-1f54fdb93a93.png">

**Comment by animosity22** [comment]
It's not needed.

```
felix@lucidity:~$ rclone config show pcloud:
--------------------
[pcloud]
type = pcloud
token = {"access_token":"key","token_type":"bearer","expiry":"0001-01-01T00:00:00Z"}
--------------------
felix@lucidity:~$ rclone ls pcloud:
 22250443 Getting started with pCloud.pdf
  6319438 My Videos/pCloud.mp4
  1442376 My Music/Demo Audio 2.mp3
  6698872 My Music/GotJoy.mp3
 28096964 My Music/Lovely Day.wav
 11252576 My Music/Momentum.mp3
   666846 My Pictures/friends.jpg
   189628 My Pictures/happy-family.jpg
    32905 My Pictures/in-the-sky.jpg
   500130 My Pictures/lovers.jpg
   238222 My Pictures/romance.jpg
   162388 My Pictures/sweet.jpg
felix@lucidity:~$ ^C
felix@lucidity:~$
```
It uses the default API if you don't provide it.

**Comment by ncw** [comment]
I think I see what is going on here.

It is in the `rclone authorize` phase that we get to know which region the pcloud is in and set the hostname but this isn't transferred back in the cut and paste part.

The hostname only needs to be set for the new european region `eapi` which is probably why you don't have it @animosity22 

There will be other remotes with this problem too I think (eg sharefile).

I'm not sure how to fix this - maybe a comment in the docs for the moment. What do you think @sylvainLS ?

**Comment by animosity22** [comment]
Assuming the person is using the European side, wouldn't they just put it in?

![image](https://user-images.githubusercontent.com/5088712/90815804-40326500-e2f9-11ea-993d-cfe304487327.png)

**Comment by ncw** [comment]
> Assuming the person is using the European side, wouldn't they just put it in?

Yes they could do. The host you are supposed to use comes back when you do the oauth.

Maybe I should just put two options in there api and eapi with a suggestion to set one if using `rclone authorize`

**Comment by animosity22** [comment]
If there are only two options, that feels like a pretty solution to the issue.

**Comment by olaulau** [comment]
I can confirm this bug still happens.
(packages from debian repositories)

headless  system : debian 11
rclone v1.53.3-DEV
- os/arch: linux/amd64
- go version: go1.15.9

gui system : debian testing
rclone v1.60.1-DEV
- os/version: debian trixie/sid (64 bit)
- os/kernel: 6.3.0-1-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.19.8
- go/linking: dynamic
- go/tags: none

**Comment by animosity22** [comment]
Those are older versions and you should not use whatever is in the repo.
